src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,44 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.model.ValidationException;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_CHARACTERS = 6;
+    private static final int MIN_AGE = 18;
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+        checkUserExists(user.getLogin());
+        return user;
+    }
+
+    private void checkUserExists(String login) {
+        if (storageDao.get(login) != null) {
+            throw new ValidationException("User with your login has already exists.Try new login!");
```

Comment:
The exception message should be more informative and should not include personal pronouns. Consider changing it to something like 'User with login [login] already exists.'